lib/streamlit/runtime/runtime.py

```diff
@@ -619,7 +613,7 @@ def _send_message(self, session_info: SessionInfo, msg: ForwardMsg) -> None:
         # Ship it off!
         session_info.client.write_forward_msg(msg_to_send)
 
-    def _enqueued_some_message(self) -> None:
+    def enqueued_some_message(self) -> None:
```

Comment:
Making this function, and `get_async_objs` below, public, feels bad. Can we have SessionManager not take a Runtime instance at all? Could `enqueued_some_message` be a signal on SessionManager that Runtime subscribes to?

Reply:
I can look into this after making some of the other suggested changes since I do have a small concern around this one.

My concern with having `enqueued_some_message` be a signal on `SessionManager` that `Runtime` subscribes to would be that it would cost a few extra event loop ticks from when we enqueue a forward message to when the coroutine responsible for sending forward messages actually does so. We'd need
1. the signal to be sent from an `AppSession` when a forward message is enqueued
2. the callback that `Runtime` has registered to this signal to fire, which sets the `need_send_data` flag
3. the coroutine to pick up that the flag has changed and loop

I'm not sure if the added step + additional few ticks (assuming that there may be some tasks queued on the event loop between steps 1 and 2) would actually mean anything in practice, but it does seem at least mildly concerning given how frequently we'd pay the additional cost in scripts that send a ton of forward messages.

I'll address `SessionManager` not taking a `Runtime` instance in another comment.

Reply:
I might be misunderstanding, but I don't think having this be an event-driven process needs to introduce any additional latency. (To be clear, I'm just talking about a `blinker.Signal`, the synchronous eventing library we use in a few other places.) Adding a callback to a Signal is essentially the same as having that callback called directly by the signal-invoking code; the benefit here would be the decoupling of SessionManager from Runtime, but the end result should be identical.

(Alternatively, we could pass the `_enqueued_some_message` function as a param to the SessionManager constructor, skip the Signal business, and still have a nice separation of concerns.)

Reply:
Ah whoops, this is due to me misunderstanding how `blinker.Signal` works. I don't think my concerns are valid in that case. I'll look into both passing `_enqueued_some_message` into the `SessionManager` constructor as well as the `Signal` approach to see which one feels better